src/ServiceControl/CustomChecks/CustomCheckResultProcessor.cs

```diff
@@ -0,0 +1,69 @@
+namespace ServiceControl.CustomChecks
+{
+    using System;
+    using System.Threading.Tasks;
+    using Contracts.CustomChecks;
+    using Infrastructure;
+    using Infrastructure.DomainEvents;
+    using NServiceBus.Logging;
+    using ServiceControl.Persistence;
+
+    class CustomCheckResultProcessor
+    {
+        public CustomCheckResultProcessor(IDomainEvents domainEvents, ICustomChecksDataStore store)
+        {
+            this.domainEvents = domainEvents;
+            this.store = store;
+        }
+
+        public async Task ProcessResult(CustomCheckDetail checkDetail)
+        {
+            try
+            {
+                var statusChange = await store.UpdateCustomCheckStatus(checkDetail).ConfigureAwait(false);
```

Comment:
Rather than return whether or not the status changed, we could just return the original status. That way the logic to determine whether or not it has changed could be container here.